src/town_type.h

```diff
@@ -91,6 +91,13 @@ enum TownLayout : uint8_t {
 };
 DECLARE_ENUM_AS_ADDABLE(TownLayout)
 
+enum class TownExpandMode : uint8_t {
+	Buildings,
+	Roads,
+};
+
```

Comment:
Could be wrong but I don't think we usually have an empty line between an enum and its modifier (`using`, `DECLARE_ENUM_AS_ADDABLE`, etc.)

```suggestion
```

Reply:
We do in some places, don't in others. The enum and the using are type definitions so should really be separated.

The bigger thing missing is doxygen.